cluster-autoscaler/provisioningrequest/service/v1beta1client/testutils.go

```diff
@@ -0,0 +1,151 @@
+	klog "k8s.io/klog/v2"
+)
+
+// NewFakeProvisioningRequestClient mock ProvisioningRequestClient for tests.
+func NewFakeProvisioningRequestClient(ctx context.Context, t *testing.T, prs ...*provreqwrapper.ProvisioningRequest) (*ProvisioningRequestClient, *FakeProvisioningRequestForceClient) {
+	t.Helper()
+	prClient := fake.NewSimpleClientset()
+	ptClient := fake_kubernetes.NewSimpleClientset()
+	for _, pr := range prs {
+		if pr == nil {
+			continue
+		}
+		if _, err := prClient.AutoscalingV1beta1().ProvisioningRequests(pr.Namespace()).Create(ctx, pr.V1Beta1(), metav1.CreateOptions{}); err != nil {
+			t.Errorf("While adding a ProvisioningRequest: %s/%s to fake client, got error: %v", pr.Namespace(), pr.Name(), err)
+		}
+		for _, pd := range pr.PodTemplates() {
+			if _, err := ptClient.CoreV1().PodTemplates(pr.Namespace()).Create(ctx, pd, metav1.CreateOptions{}); err != nil {
+				t.Errorf("While adding a PodTemplate: %s/%s to fake client, got error: %v", pr.Namespace(), pd.Name, err)
+			}
+		}
+	}
+	prLister, err := newAllPRsLister(prClient, make(chan struct{}))
+	if err != nil {
+		t.Fatalf("Failed to create Provisioning Request lister. Error was: %v", err)
+	}
+	ptLister, err := newFakePodTemplatesLister(t, ptClient, make(chan struct{}))
+	if err != nil {
+		t.Fatalf("Failed to create Provisioning Request lister. Error was: %v", err)
+	}
+	return &ProvisioningRequestClient{
+			client:   prClient,
+			prLister: prLister,
+			ptLister: ptLister,
+		}, &FakeProvisioningRequestForceClient{
+			client: prClient,
+		}
+}
+
+// FakeProvisioningRequestForceClient that allows to skip cache.
+type FakeProvisioningRequestForceClient struct {
+	client *fake.Clientset
+}
+
+// ProvisioningRequest gets a specific ProvisioningRequest CR, skipping cache.
+func (c *FakeProvisioningRequestForceClient) ProvisioningRequest(namespace, name string) (*v1beta1.ProvisioningRequest, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), provisioningRequestClientCallTimeout)
+	defer cancel()
+	return c.client.AutoscalingV1beta1().ProvisioningRequests(namespace).Get(ctx, name, metav1.GetOptions{})
+}
+
+// newFakePodTemplatesLister creates a fake lister for the Pod Templates in the cluster.
+func newFakePodTemplatesLister(t *testing.T, client kubernetes.Interface, channel <-chan struct{}) (v1.PodTemplateLister, error) {
+	t.Helper()
+	factory := informers.NewSharedInformerFactory(client, 1*time.Hour)
+	ptLister := factory.Core().V1().PodTemplates().Lister()
+	factory.Start(channel)
+	informersSynced := factory.WaitForCacheSync(channel)
+	for _, synced := range informersSynced {
+		if !synced {
+			return nil, fmt.Errorf("can't create Pod Template lister")
+		}
+	}
+	klog.V(2).Info("Successful initial Pod Template sync")
+	return ptLister, nil
+}
+
+// ProvisioningRequestBetaForTests returns test ProvisioningRequest.
+func ProvisioningRequestBetaForTests(namespace, name string) *provreqwrapper.ProvisioningRequest {
```

Comment:
Is there a reason for this method to be public? If it is meant to be reused in multiple packages, then `provreqwrapper` package seems like a better place for it.